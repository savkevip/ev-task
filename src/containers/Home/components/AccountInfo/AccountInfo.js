import React from "react";
import PropTypes from "prop-types";
import { AccountInfoWrapper } from "./styles/style";

const AccountInfo = ({ name, iban, balance }) => (
  <AccountInfoWrapper>
    <h1>{name}</h1>
    <h3>iban: {iban}</h3>
    <h3>balance: {balance}</h3>
  </AccountInfoWrapper>
);

AccountInfo.propTypes = {
  name: PropTypes.string,
  iban: PropTypes.string,
  balance: PropTypes.number
};

export default AccountInfo;
