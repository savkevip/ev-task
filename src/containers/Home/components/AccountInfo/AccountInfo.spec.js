import React from "react";
import { shallow } from "enzyme";
import AccountInfo from "./AccountInfo";

const account = {
  name: "Jack Torrance",
  iban: "123456789",
  balance: 500
};

describe("AccountInfo", () => {
  it("should match snapshot", function() {
    const component = shallow(
      <AccountInfo
        name={account.name}
        iban={account.iban}
        balance={account.balance}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
