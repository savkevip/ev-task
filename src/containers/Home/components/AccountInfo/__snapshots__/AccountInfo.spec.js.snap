// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfo should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountInfo
    balance={500}
    iban="123456789"
    name="Jack Torrance"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h1>
          Jack Torrance
</h1>,
        <h3>
          iban: 
          123456789
</h3>,
        <h3>
          balance: 
          500
</h3>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jack Torrance",
        },
        "ref": null,
        "rendered": "Jack Torrance",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "iban: ",
            "123456789",
          ],
        },
        "ref": null,
        "rendered": Array [
          "iban: ",
          "123456789",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "balance: ",
            500,
          ],
        },
        "ref": null,
        "rendered": Array [
          "balance: ",
          500,
        ],
        "type": "h3",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h1>
            Jack Torrance
</h1>,
          <h3>
            iban: 
            123456789
</h3>,
          <h3>
            balance: 
            500
</h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jack Torrance",
          },
          "ref": null,
          "rendered": "Jack Torrance",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "iban: ",
              "123456789",
            ],
          },
          "ref": null,
          "rendered": Array [
            "iban: ",
            "123456789",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "balance: ",
              500,
            ],
          },
          "ref": null,
          "rendered": Array [
            "balance: ",
            500,
          ],
          "type": "h3",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
